scripts/generate-clients/copy-to-clients.js

```diff
@@ -62,6 +67,24 @@ const mergeManifest = (fromContent, toContent) => {
   return merged;
 };
 
+/**
+ * Remove "^" from the the version of dependencies on @aws-sdk packages.
+ * e.g. "@aws-sdk/config-resolver": "^1.0.0-gamma.0"
+ *      => "@aws-sdk/config-resolver": "1.0.0-gamma.0"
+ */
+const pinDependencies = manifest => {
+  const removeRangeVersion = ([name, version]) =>
+    name.indexOf("@aws-sdk/") === 0
+      ? [name, version.replace("^", "")]
+      : [name, version];
+  manifest.dependencies = Object.fromEntries(
+    Object.entries(manifest.dependencies).map(removeRangeVersion)
+  );
+  manifest.devDependencies = Object.fromEntries(
+    Object.entries(manifest.devDependencies).map(removeRangeVersion)
+  );
+};
```

Comment:
Optional: use functions all the way to avoid mutating manifest and having small functions 

```js
const filterAwsSdkDeps = (name) => name.indexOf("@aws-sdk/") === 0;
const removeCaretFromDepVersion = (version) => version.replace("^", "");

const getPinnedAwsSdkDeps = (deps) => ({
  ...deps,
  ...Object.entries(deps)
    .filter(([name, version]) => filterAwsSdkDeps(name))
    .reduce(
      (acc, [name, version]) => ({
        ...acc,
        [name]: removeCaretFromDepVersion(version),
      }),
      {}
    ),
});

const getManifestWithPinnedAwsSdkDeps = (manifest) => ({
  ...manifest,
  dependencies: getPinnedAwsSdkDeps(manifest.dependencies),
  devDependencies: getPinnedAwsSdkDeps(manifest.devDependencies),
});
```